packages/utils/upgrade/src/core/version.ts

```diff
@@ -1,15 +1,41 @@
 import * as semver from 'semver';
 
 export type SemVer = `${number}.${number}.${number}`;
-export type LatestVersion = 'latest';
 
-export type AnyVersion = SemVer | LatestVersion;
+export enum VersionRelease {
+  Latest = 'latest',
+  Major = 'major',
+  Minor = 'minor',
+  Patch = 'patch',
+}
+
+export type Version = SemVer | VersionRelease;
 
 export interface VersionRange {
   from: SemVer;
-  to: AnyVersion;
+  to: Version;
 }
 
+export const isVersionRelease = (version: string): version is VersionRelease => {
+  return Object.values<string>(VersionRelease).includes(version);
+};
+
+export const isLatestVersion = (str: string): str is VersionRelease.Latest => {
+  return str === VersionRelease.Latest;
+};
+
+export const isVersion = (str: string): str is Version => {
+  return isVersionRelease(str) || isSemVer(str);
+};
+
+export const isSemVer = (str: string): str is SemVer => {
```

Comment:
I would just rely on semver methods like `semver.valid(str)` and we can cut out this code and also handle pre-release tags (almost certainly will never release them, but might have some use for testing?)

There are too many weird edge cases to trust writing our own code